precompile/fee_config_manager.go

```diff
@@ -0,0 +1,268 @@
+		feeConfig.MaxBlockGasCost.FillBytes(make([]byte, 32)),
+		feeConfig.BlockGasCostStep.FillBytes(make([]byte, 32)),
+	}
+	if useSelector {
+		packed = append([][]byte{setFeeConfigSignature}, packed...)
+		return packOrderedHashesWithSelector(packed, fullLen+selectorLen)
+	}
+
+	return packOrderedHashes(packed, fullLen)
+}
+
+// UnpackFeeConfigInput attempts to unpack [input] into the arguments to the fee config precompile
+// assumes that [input] does not include selector (omits first 4 bytes in PackSetFeeConfigInput)
+func UnpackFeeConfigInput(input []byte) (commontype.FeeConfig, error) {
+	if len(input) != feeConfigInputLen {
+		return commontype.FeeConfig{}, fmt.Errorf("invalid input length for fee config input: %d", len(input))
+	}
+	return commontype.FeeConfig{
+		GasLimit:                 new(big.Int).SetBytes(returnPackedElement(input, 0)),
+		TargetBlockRate:          new(big.Int).SetBytes(returnPackedElement(input, 1)).Uint64(),
+		MinBaseFee:               new(big.Int).SetBytes(returnPackedElement(input, 2)),
+		TargetGas:                new(big.Int).SetBytes(returnPackedElement(input, 3)),
+		BaseFeeChangeDenominator: new(big.Int).SetBytes(returnPackedElement(input, 4)),
+		MinBlockGasCost:          new(big.Int).SetBytes(returnPackedElement(input, 5)),
+		MaxBlockGasCost:          new(big.Int).SetBytes(returnPackedElement(input, 6)),
+		BlockGasCostStep:         new(big.Int).SetBytes(returnPackedElement(input, 7)),
+	}, nil
+}
+
+// GetStoredFeeConfig returns fee config from contract storage in given state
+func GetStoredFeeConfig(stateDB StateDB) (commontype.FeeConfig, bool) {
+	if !stateDB.Exist(FeeConfigManagerAddress) {
+		return commontype.FeeConfig{}, false
+	}
+	feeConfig := commontype.FeeConfig{}
+	isAllZeros := true
+	for i := minFeeConfigFieldKey; i <= maxFeeConfigFieldKey; i++ {
+		val := stateDB.GetState(FeeConfigManagerAddress, common.Hash{byte(i)})
+		// even if one of these vals are non-zero means that this state exist
+		if isAllZeros && val != (common.Hash{0}) {
+			isAllZeros = false
+		}
+		switch i {
+		case gasLimitKey:
+			feeConfig.GasLimit = new(big.Int).Set(val.Big())
+		case targetBlockRateKey:
+			feeConfig.TargetBlockRate = val.Big().Uint64()
+		case minBaseFeeKey:
+			feeConfig.MinBaseFee = new(big.Int).Set(val.Big())
+		case targetGasKey:
+			feeConfig.TargetGas = new(big.Int).Set(val.Big())
+		case baseFeeChangeDenominatorKey:
+			feeConfig.BaseFeeChangeDenominator = new(big.Int).Set(val.Big())
+		case minBlockGasCostKey:
+			feeConfig.MinBlockGasCost = new(big.Int).Set(val.Big())
+		case maxBlockGasCostKey:
+			feeConfig.MaxBlockGasCost = new(big.Int).Set(val.Big())
+		case blockGasCostStepKey:
+			feeConfig.BlockGasCostStep = new(big.Int).Set(val.Big())
+		default:
+			panic("unknown key")
+		}
+	}
+	if isAllZeros {
+		return commontype.EmptyFeeConfig, false
+	}
+	return feeConfig, true
+}
+
+func StoreFeeConfig(stateDB StateDB, feeConfig commontype.FeeConfig) {
```

Comment:
nit: should we verify the byte size of each val in feeConifg is less than 32 bytes? (otherwise common.BigToHash crops left)

Reply:
it's actually safe since this is being used after we unpack it with HashLength from inputs:
https://github.com/ava-labs/subnet-evm/blob/fee-config-precompile/precompile/fee_config_manager.go#L213

but wouldn't hurt adding another check